common-test/src/main/java/org/alfresco/hxi_connector/common/test/util/DockerContainers.java

```diff
@@ -0,0 +1,151 @@
+/*-
+ * #%L
+ * Alfresco HX Insight Connector
+ * %%
+ * Copyright (C) 2023 - 2024 Alfresco Software Limited
+ * %%
+ * This file is part of the Alfresco software.
+ * If the software was purchased under a paid Alfresco license, the terms of
+ * the paid license agreement will prevail.  Otherwise, the software is
+ * provided under the following open source license terms:
+ *
+ * Alfresco is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Alfresco is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Alfresco. If not, see <http://www.gnu.org/licenses/>.
+ * #L%
+ */
+package org.alfresco.hxi_connector.common.test.util;
+
+import static java.util.function.Predicate.not;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.Duration;
+import java.util.Locale;
+import java.util.Optional;
+import java.util.function.Predicate;
+import java.util.stream.Stream;
+
+import lombok.AccessLevel;
+import lombok.Cleanup;
+import lombok.NoArgsConstructor;
+import lombok.SneakyThrows;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.containers.wait.strategy.Wait;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.utility.DockerImageName;
+
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public class DockerContainers
+{
+    private static final String REPOSITORY_IMAGE = "alfresco/alfresco-content-repository-community";
+    private static final String REPOSITORY_TAG = DockerTags.getRepositoryTag();
+    private static final String POSTGRES_IMAGE = "postgres";
+    private static final String POSTGRES_TAG = DockerTags.getPostgresTag();
+    private static final String ACTIVE_MQ_IMAGE = "quay.io/alfresco/alfresco-activemq";
+    private static final String ACTIVE_MQ_TAG = DockerTags.getActiveMqTag();
+    private static final String DB_USER = "alfresco";
+    private static final String DB_PASS = "alfresco";
+    private static final String DB_NAME = "alfresco";
+    private static final String REPOSITORY_ALIAS = "repository";
+    private static final String POSTGRES_ALIAS = "postgres";
+    private static final String ACTIVE_MQ_ALIAS = "activemq";
+
+    public static GenericContainer<?> createExtendedRepositoryContainerWithin(Network network)
```

Comment:
Did you try with https://java.testcontainers.org/modules/docker_compose/ approach?
I am not saying it's better but it might make use of docker compose files we already have in `distribution` module. On the other hand, I am not sure how docker tags from generated `.env` file would be handled,

Reply:
I'm not in favor of using d-c file - it's not so flexible as programmatic containers - not all services are required while running tests. Also configuration of one service can vary for different tests. TBH I didn't tried it - that might work, but I expect it will introduce some extra issues on the way 🤔